pkg/processor/reportprocessor.go

```diff
@@ -191,68 +191,147 @@ func getGovernanceResults(dynamicClient dynamic.Interface, clusterInfo types.Man
-							Result:      "fail",
-							Properties: map[string]string{
-								"created_at": md["creationTimestamp"].(string),
-								"total_risk": convertSevFromGovernance(getSevFromTemplate(plc, templateMeta["name"].(string))),
-							},
-						})
-					}
-				}
+		// Parse relevant policy fields
+		plcName := plc.GetName()
+		category := plc.GetAnnotations()["policy.open-cluster-management.io/categories"]
+		md, _, err := unstructured.NestedMap(plc.Object, "metadata")
+		if err != nil {
+			glog.Warningf("error parsing metadata as a map for policy %s: %s", plcName, err)
+			continue
+		}
+		status, _, err := unstructured.NestedMap(plc.Object, "status")
+		if err != nil {
+			glog.Warningf("error parsing status as a map for policy %s: %s", plcName, err)
+			continue
+		}
+		details, _, err := unstructured.NestedSlice(status, "details")
+		if err != nil {
+			glog.Warningf("error parsing status details as a map for policy %s: %s", plcName, err)
+			continue
+		}
+
+		// Only create a violation if the policy is non-compliant
+		if status["compliant"].(string) != "NonCompliant" {
+			continue
+		}
+
+		// Generate a cluster policy report result for each non-compliant policy template within the root policy
+		for idx, detail := range details {
+			var detailMap map[string]interface{}
+			var ok bool
+			if detailMap, ok = detail.(map[string]interface{}); !ok {
+				glog.Warningf("failed to parse history detail as a map for policy %s, template %d", plcName, idx)
+				continue
 			}
-		}()
+
+			// Only create a violation if the policy template is non-compliant
+			if detailMap["compliant"] != "NonCompliant" {
+				continue
+			}
+
+			// Parse relevant status fields for the policy template
+			plcTemplateName, _, err := unstructured.NestedString(detailMap, "templateMeta", "name")
+			if err != nil {
+				glog.Warningf(
+					"error parsing policy template name as a string for policy %s, template %d: %s", plcName, idx, err)
+				continue
+			}
+			historyItems, _, err := unstructured.NestedSlice(detailMap, "history")
+			glog.V(1).Infof("error parsing history as a slice for policy %s, template %d: %s", plcName, idx, err)
+			if len(historyItems) == 0 {
+				glog.Warningf("history is empty for policy %s, template %d", plcName, idx)
+				continue
+			}
+			message, _, err := unstructured.NestedString(historyItems[0].(map[string]interface{}), "message")
+			if err != nil {
+				glog.Warningf(
+					"error parsing compliance message as a string for policy %s, template %d: %s", plcName, idx, err)
+				continue
+			}
+
+			// Append violation to policy report results
+			clusterViolations = append(clusterViolations, &v1alpha2.PolicyReportResult{
+				Policy:      plcName,
+				Description: message,
+				Scored:      false,
+				Category:    category,
+				Source:      "grc",
+				Timestamp:   metav1.Timestamp{Seconds: time.Now().Unix(), Nanos: int32(time.Now().UnixNano())},
```

Comment:
We will leave it as is in this PR, but it seems to me that this should be the timestamp in the policy history.

Reply:
Issue filed: https://issues.redhat.com/browse/ACM-6921